README.md

```diff
@@ -606,6 +606,81 @@ Output is written to stdout.
       future obfuscation with --load_map.); default: "";
 ```
 
+## Indexing Verible C++ Code using Kythe
+
+The steps mentioned here can be generalized for indexing Bazel-based projects.
+
+### Initializing Kythe
+Download the latest Kythe release from https://github.com/kythe/kythe/releases and then unpack it for a snapshot of Kythe’s toolset.
+
+```bash
+tar xzf kythe-v*.tar.gz
+rm -rf /opt/kythe
+mv kythe-v*/ /opt/kythe
+```
+More information can be found [here](https://github.com/kythe/kythe#getting-started).
+
+Clone Kythe from [here](https://github.com/kythe/kythe). Then from within the Kythe clone, build the web frontend and copy its files into /opt/kythe/
+
+```bash
+bazel build //kythe/web/ui
+mkdir -p /opt/kythe/web/ui
+cp -r bazel-bin/kythe/web/ui/resources/public/* /opt/kythe/web/ui
+cp -r kythe/web/ui/resources/public/* /opt/kythe/web/ui
+chmod -R 755 /opt/kythe/web/ui
+```
+More information can be found [here](https://github.com/google/haskell-indexer#building-from-source).
+
+### Extracting Compilations for Verible
+
+#### VNames.json File
+`vnames.json` file is used for renaming certain filepaths during extraction, but renaming isn’t needed here but you can add the renaming you find suitable.
+
+#### Run the extractor
+```bash
+# run on all targets
+bazel test --experimental_action_listener=:extract_cxx  //...
+
+# run on specific target (e.g. some cc_binary)
```

Comment:
cc_binary or cc_library?

Reply:
I tried it on cc_binary and cc_library and it worked and I believe it works with bazel-targets in general.